[BUG] [BUG] Scikit-learn 1.7.0 has a bug for generating HTML representation

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating system

- [x] Linux
- [ ] Mac
- [ ] Windows


### Python version

- [ ] 3.14
- [x] 3.13
- [ ] 3.12
- [ ] 3.11
- [ ] 3.10

### nilearn version

0.13.2.dev134+g356bd6b23

### scikit-learn version

1.7.0

### Expected behavior

`nilearn.maskers/tests/test_html_report.py::test_multi_nifti_masker_generate_report_mask` should pass.

### Current behavior & error messages

This is what I got:



```bash
...

>       if param_value != init_default_params[param_name] and not (
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            is_scalar_nan(init_default_params[param_name])
            and is_scalar_nan(param_value)
        ):
E       ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

init_default_params = {'clean_args': None, 'cmap': 'gray', 'detrend': False, 'dtype': None, ...}
param_name = 'target_affine'
param_value = array([[1., 0., 0., 0.],
       [0., 1., 0., 0.],
       [0., 0., 1., 0.],
       [0., 0., 0., 1.]])



```


### Steps and code to reproduce bug

Run test from command line
```bash
pytest nilearn/maskers/tests/test_html_report.py::test_multi_nifti_masker_generate_report_mask

```


### Solution

The bug is fixed in scikit-learn version [1.7.1](https://scikit-learn.org/stable/whats_new/v1.7.html#version-1-7-1).

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,7 +37,10 @@ min = [
     "numpy==1.26.0",
     "pandas==2.3.0",
     "requests==2.30.0",
-    "scikit-learn==1.5.0",
+    # scikit-learn version 1.7.0 has a bug that prevents _repr_html_ to run
+    # correctly
+    # https://scikit-learn.org/stable/whats_new/v1.7.html#sklearn-base
+    "scikit-learn==1.5.0,!=1.7.0",
     "scipy==1.11.1",
     "jinja2==3.1.2"
 ]

